core/src/main/java/io/grpc/internal/Channelz.java

```diff
@@ -33,7 +33,7 @@
       new ConcurrentHashMap<Long, Instrumented<ServerStats>>();
   private final ConcurrentMap<Long, Instrumented<ChannelStats>> rootChannels =
       new ConcurrentHashMap<Long, Instrumented<ChannelStats>>();
-  private final ConcurrentMap<Long, Instrumented<ChannelStats>> channels =
+  private final ConcurrentMap<Long, Instrumented<ChannelStats>> subchannels =
```

Comment:
why not keep them all in the same list?

Reply:
I think we won't want a single navigable concurrent map for all channels and subchannels. We want to be fast when looking for all root channels and avoid having to scan through subchannels. Since we can't get rid of the special root channels navigable map, and the fact that getChannel is different from getSubchannel, I think mixing them won't give us much.